### PR TITLE
WIP Update to bytes

### PR DESCRIPTION
**Pull request by corchessergiu**

### CODE REVIEW

File: contracts/Deb0x.sol [hunk 1]
```diff
@@ -595,21 +680,29 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
         bytes32 selfBodyHash = keccak256(
             abi.encodePacked(crefs[recipients.length - 1])
         );
+        require(crefs[recipients.length - 1].length <= 8 , "Deb0x: crefs too long");
```
Comment (dappcoder) [inline contracts/Deb0x.sol]: change "crefs" to "cref" in error message

File: contracts/Deb0x.sol [hunk 2]
```diff
@@ -575,13 +647,26 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
...
+     * @param crefs content references to the messages.
+     */
+    function _send(address[] memory recipients, bytes32[][] memory crefs)
         internal
         returns (uint256)
     {
+        require(recipients.length == crefs.length, "Deb0x: Arrays must have same length");
+        require(recipients.length > 0, "Deb0x: Length of arrays must be greater than 0");
+        for (uint256 idx = 0; idx < recipients.length - 1; idx++) {
+            require(crefs[recipients.length - 1].length <= 8 , "Deb0x: crefs too long");
```
Comment (dappcoder) [inline contracts/Deb0x.sol]: change message to "Deb0x: one of crefs is too long"

File: contracts/Deb0x.sol [hunk 3]
```diff
@@ -575,13 +647,26 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
...
+     * Lengths of recipients and crefs arrays must match.
+     * 
+     * @param recipients recipient addresses that messages are stored for.
+     * @param crefs content references to the messages.
+     */
+    function _send(address[] memory recipients, bytes32[][] memory crefs)
         internal
         returns (uint256)
     {
+        require(recipients.length == crefs.length, "Deb0x: Arrays must have same length");
```
Comment (dappcoder) [inline contracts/Deb0x.sol]: "Deb0x: crefs and recipients lengths not equal"

File: contracts/Deb0x.sol [hunk 4]
```diff
@@ -575,13 +647,26 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
...
+        require(recipients.length > 0, "Deb0x: Length of arrays must be greater than 0");
```
Comment (dappcoder) [inline contracts/Deb0x.sol]: "Deb0x: recipients array empty"

File: contracts/Deb0xAttacker.sol [hunk 5]
```diff
@@ -29,6 +29,6 @@ contract Deb0xAttacker {
 
         tempPayload[0] = "0x";
         tempPayload[1] = "0x";
-        deb0xInstance.send{value: 100000000 gwei}(tempTo, tempPayload, 0x0000000000000000000000000000000000000000, 0 ,0);
+      //  deb0xInstance.send{value: 100000000 gwei}(tempTo, tempPayload, 0x0000000000000000000000000000000000000000, 0 ,0);
```
Comment (dappcoder) [inline contracts/Deb0xAttacker.sol]: why commented?

**Review by dappcoder — CHANGES_REQUESTED** [review]

File: contracts/Deb0x.sol [hunk 6]
```diff
@@ -57,14 +65,29 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
 
     mapping(address => uint256) public clientAccruedFees;
 
+    /**
+     * ????  why  0? when getUnclaimedRewards is >0 ?
```
Comment (CatalinBalut) [inline contracts/Deb0x.sol]: ?

File: contracts/Deb0x.sol [hunk 7]
```diff
@@ -57,14 +65,29 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
...
+     * Current unclaimed rewards and staked amounts per account.
+     */
     mapping(address => uint256) public accRewards;
 
+    /**
+     * ?
```
Comment (CatalinBalut) [inline contracts/Deb0x.sol]: ?

File: contracts/Deb0x.sol [hunk 8]
```diff
@@ -133,17 +156,20 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
...
         bytes32 indexed hash,
         uint256 sentId,
         uint256 timestamp,
-        string content
+        bytes32[] content
     );
 
-
-    event KeySet(address indexed to, bytes32 indexed hash, string value);
+    event KeySet(
```
Comment (CatalinBalut) [inline contracts/Deb0x.sol:164]: can be formatted to 1 line of code

File: contracts/Deb0x.sol [hunk 9]
```diff
@@ -207,18 +252,20 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
         nonReentrant()
         gasWrapper(nativeTokenFee)
         gasUsed(feeReceiver, msgFee)
+
     {
+        require(msgFee < 10001, "Deb0x: Reward fees can not exceed 100%");
+
+
```
Comment (CatalinBalut) [inline contracts/Deb0x.sol]: 258 can be deleted

File: contracts/Deb0x.sol [hunk 10]
```diff
@@ -575,13 +647,26 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
...
+     * Lengths of recipients and crefs arrays must match.
+     * 
+     * @param recipients recipient addresses that messages are stored for.
+     * @param crefs content references to the messages.
+     */
+    function _send(address[] memory recipients, bytes32[][] memory crefs)
         internal
         returns (uint256)
     {
+        require(recipients.length == crefs.length, "Deb0x: crefs and recipients lengths not equal");
```
Comment (CatalinBalut) [inline contracts/Deb0x.sol:654]: Deb0x: crefs and recipients lengths must be equal

File: contracts/Deb0x.sol [hunk 11]
```diff
@@ -575,13 +647,26 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
...
+        require(recipients.length > 0, "Deb0x: recipients array empty");
```
Comment (CatalinBalut) [inline contracts/Deb0x.sol:655]: one of arrays is empty

File: contracts/Deb0x.sol [hunk 12]
```diff
@@ -595,21 +680,29 @@ contract Deb0x is ERC2771Context, ReentrancyGuard {
         bytes32 selfBodyHash = keccak256(
             abi.encodePacked(crefs[recipients.length - 1])
         );
+        require(crefs[recipients.length - 1].length <= 8 , "Deb0x: cref too long");
```
Comment (CatalinBalut) [inline contracts/Deb0x.sol:676]: cref size is bigger than 8 bytes32

File: test/Deb0x/test_bytes32.js [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+            feeReceiver.address, 100, 0, { value: ethers.utils.parseEther("1") })
+
+        const sentEvent = await rewardedAlice.queryFilter("Sent");
+        for (let entry of sentEvent) {
+            arguments.push(ethers.utils.arrayify(entry.args.content[0]))
+            arguments.push(ethers.utils.arrayify(ethers.utils.stripZeros(entry.args.content[1])))
+            console.log(Converter.convertBytes32ToString(Array.from((arguments))));
+            arguments = [];
+        }
+    });
```
Comment (CatalinBalut) [inline test/Deb0x/test_bytes32.js:80]: must have some expect to check the result

File: test/Deb0x/test_bytes32.js [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
```
Comment (CatalinBalut) [inline test/Deb0x/test_bytes32.js:81]: +1 test for expect event

File: test/Deb0x/test_claimFeesStakeAndUnstake.js [hunk 15]
```diff
@@ -27,12 +30,12 @@ describe("Test claimFeesStakeAndUnstake", async function() {
...
-        await rewardedBob["send(address[],string[],address,uint256,uint256)"]([messageReceiver.address], ["ipfs://"], ethers.constants.AddressZero, 0, 0, { value: ethers.utils.parseEther("1") })
-        await rewardedBob["send(address[],string[],address,uint256,uint256)"]([messageReceiver.address], ["ipfs://"], ethers.constants.AddressZero, 0, 0, { value: ethers.utils.parseEther("1") })
-        await rewardedCarol["send(address[],string[],address,uint256,uint256)"]([messageReceiver.address], ["ipfs://"], ethers.constants.AddressZero, 0, 0, { value: ethers.utils.parseEther("1") })
-        await rewardedCarol["send(address[],string[],address,uint256,uint256)"]([messageReceiver.address], ["ipfs://"], ethers.constants.AddressZero, 0, 0, { value: ethers.utils.parseEther("1") })
-        await rewardedCarol["send(address[],string[],address,uint256,uint256)"]([messageReceiver.address], ["ipfs://"], ethers.constants.AddressZero, 0, 0, { value: ethers.utils.parseEther("1") })
+        await rewardedAlice["send(address[],bytes32[][],address,uint256,uint256)"]([messageReceiver.address], [payload], ethers.constants.AddressZero, 0, 0, { value: ethers.utils.parseEther("1") })
```
Comment (CatalinBalut) [inline test/Deb0x/test_claimFeesStakeAndUnstake.js:33]: optional: the payload is named crefs/cref

File: test/Deb0x/test_setKeyWithBytes32.js [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+
+        const dbxAddress = await rewardedAlice.dbx()
+        dbxERC20 = new ethers.Contract(dbxAddress, abi, hre.ethers.provider)
+
+        rewardedBob = rewardedAlice.connect(bob)
+        rewardedCarol = rewardedAlice.connect(carol)
+        frontend = rewardedAlice.connect(feeReceiver)
+    });
+
+    it("Only for check", async() => {
```
Comment (CatalinBalut) [inline test/Deb0x/test_setKeyWithBytes32.js:24]: Should have a better messege like "Should check if the publicKey is stored in the correct format on Deb0x.sol"

File: interface/src/components/App/Encrypt.tsx [hunk 17]
```diff
@@ -132,6 +133,7 @@ export function Encrypt(replyAddress: any): any {
 
     async function isInitialized(address: any) {
         const deb0xViewsContract = deb0xViews(library, deb0xViewsAddress);
+        console.log(await getKey(address))
```
Comment (CatalinBalut) [inline interface/src/components/App/Encrypt.tsx]: delete console.log

File: interface/src/components/App/Encrypt.tsx [hunk 18]
```diff
@@ -270,6 +273,7 @@ export function Encrypt(replyAddress: any): any {
                 await fetchSendResult(request, url)
 
             } catch (error: any) {
+                console.log(error)
```
Comment (CatalinBalut) [inline interface/src/components/App/Encrypt.tsx]: delete console.log

**Review by CatalinBalut — CHANGES_REQUESTED** [review]

**Review by CatalinBalut — APPROVED** [review]

**Pull request merged**